src/lib.rs

```diff
@@ -565,8 +614,71 @@ pub fn get_lines_from_osm(pbf: &mut OsmPbfReader) -> Vec<Line> {
         .collect()
 }
 
+pub fn get_routes_from_stop(routes: &Vec<Route>, stop_point: &StopPoint) -> Vec<Route> {
+    routes
+        .iter()
+        .filter(|route| route.contains_stop_point_id(&stop_point.id))
+        .map(|obj| obj.clone())
+        .collect()
+}
+
+pub fn categorize_stop_point(stop_point: &StopPoint, routes: Vec<Route>) -> StopPoint {
```

Comment:
Instead of cloning the `StopPoint`, it could be taken as a mutable reference.
```suggestion
pub fn categorize_stop_point(stop_point: &mut StopPoint, routes: Vec<Route>) {
```

See https://github.com/prhod/osm-transit-extractor/pull/1.